### PR TITLE
Add Vial cask

### PR DESCRIPTION
**Pull request by Technofrikus**
Added Vial as a cask, v0.3, has livecheck

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for a [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference). -- used the generate task token script
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask). -- I hope I understood everything correctly, but this is a stable version, even though it is still Beta.
- [x] `brew audit --new-cask <cask>` worked successfully.
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

### CODE REVIEW

File: Casks/vial.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+cask "vial" do
+  version "0.3"
+  sha256 "7f8861cce8782afa5f68267e3b28293517cd8f0aee8b70f621b8c208b79b5cb1"
+
+  url "https://github.com/vial-kb/vial-gui/releases/download/v#{version}/Vial-v#{version}.dmg", verified: "https://github.com/vial-kb/vial-gui"
```
Comment (SMillerDev) [inline Casks/vial.rb:5]:
```suggestion
  url "https://github.com/vial-kb/vial-gui/releases/download/v#{version}/Vial-v#{version}.dmg",
        verified: "https://github.com/vial-kb/vial-gui"
```

File: Casks/vial.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  name "vial"
+  desc "Configurator of compatible keyboards in real time"
+  homepage "https://get.vial.today/"
+
+  livecheck do
+    url "https://github.com/vial-kb/vial-gui.git"
+  end
```
Comment (SMillerDev) [inline Casks/vial.rb:12]: This is the default

**Review by SMillerDev — COMMENTED** [review]
Please add a [`zap` stanza](https://docs.brew.sh/Cask-Cookbook#stanza-zap) here. The [CreateZap](https://github.com/nrlquaker/homebrew-createzap) tool can help you with this.

**Comment by Technofrikus** [comment]
> Please add a [`zap` stanza](https://docs.brew.sh/Cask-Cookbook#stanza-zap) here. The [CreateZap](https://github.com/nrlquaker/homebrew-createzap) tool can help you with this.

Not sure if I have to comment. I added the zap stanza.

**Comment by bevanjkay** [comment]
@Technofrikus Can you try to fix the style errors? Running `brew style --fix vial` in your branch should fix most of them, and flag any that can't be fixed automatically.

**Comment by Technofrikus** [comment]
It seems the only problem now is that it is not stable version, just beta. But Vial can be considered stable in my experience and there is not stable version. So it should be added to the main repo nonetheless in my opinion.

**Comment by bevanjkay** [comment]
@Technofrikus Pending feedback from other maintainers, #108050 should allow the CI to pass

File: Casks/vial.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  version "0.3"
+  sha256 "7f8861cce8782afa5f68267e3b28293517cd8f0aee8b70f621b8c208b79b5cb1"
+
+  url "https://github.com/vial-kb/vial-gui/releases/download/v#{version}/Vial-v#{version}.dmg",
+      verified: "https://github.com/vial-kb/vial-gui"
+  name "vial"
+  desc "Configurator of compatible keyboards in real time"
+  homepage "https://get.vial.today/"
+
+  livecheck do
```
Comment (vitorgalvao) [inline Casks/vial.rb:11]: Why did you readd this? I removed it for a reason. As @SMillerDev said, this is the default, it’s redundant in this case and works without it.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

File: Casks/vial.rb (hunk 3 above)
Comment (Technofrikus) [reply Casks/vial.rb:11]: I am new to really committing on github and find the interface and system pretty complicated. Wasnt on purpose, sorry.  And honestly dont know what you or SMillerDev are referring to. I cant see a difference to my livecheck code.

Comment (bevanjkay) [reply Casks/vial.rb:11]: @technofrikus The livecheck code isn't necessary when the download URL is from Github. It is inferred from the URL.

**Review by vitorgalvao — DISMISSED** [review]

**Comment by vitorgalvao** [comment]
Verified locally. Merging due to https://github.com/Homebrew/homebrew-cask/pull/108050#issuecomment-873408387.

**Pull request merged**